M/MPItrampoline/build_tarballs.jl

```diff
@@ -155,39 +162,31 @@ cd $WORKSPACE/srcdir/MPIwrapper-*
 mkdir build
 cd build
 # Yes, this is tedious. No, without being this explicit, cmake will
-# not properly auto-detect the MPI libraries.
-if [ -f ${prefix}/lib/mpich/lib/libpmpi.${dlext} ]; then
+# not properly auto-detect the MPI libraries on Darwin.
+if [[ "${target}" == *-apple-* ]]; then
+    ext='a'
     cmake \
         -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
         -DCMAKE_FIND_ROOT_PATH="${prefix}/lib/mpich;${prefix}" \
         -DCMAKE_INSTALL_PREFIX=${prefix} \
         -DBUILD_SHARED_LIBS=ON \
-        -DMPI_C_COMPILER=cc \
         -DMPI_CXX_COMPILER=c++ \
         -DMPI_Fortran_COMPILER=gfortran \
         -DMPI_CXX_LIB_NAMES='mpicxx;mpi;pmpi' \
         -DMPI_Fortran_LIB_NAMES='mpifort;mpi;pmpi' \
-        -DMPI_mpi_LIBRARY=${prefix}/lib/mpich/lib/libmpi.${dlext} \
-        -DMPI_mpicxx_LIBRARY=${prefix}/lib/mpich/lib/libmpicxx.${dlext} \
-        -DMPI_mpifort_LIBRARY=${prefix}/lib/mpich/lib/libmpifort.${dlext} \
-        -DMPI_pmpi_LIBRARY=${prefix}/lib/mpich/lib/libpmpi.${dlext} \
+        -DMPI_pmpi_LIBRARY=${prefix}/lib/mpich/lib/libpmpi.${ext} \
+        -DMPI_mpi_LIBRARY=${prefix}/lib/mpich/lib/libmpi.${ext} \
+        -DMPI_mpicxx_LIBRARY=${prefix}/lib/mpich/lib/libmpicxx.${ext} \
+        -DMPI_mpifort_LIBRARY=${prefix}/lib/mpich/lib/libmpifort.${ext} \
```

Comment:
Just to understand: why are you using the static libraries?  So that you don't need to ship the shared libraries around?

Reply:
MPItrampoline comes with a fall-back MPICH that is wrapped by MPIwrapper. This MPIwrapper has problems loading the respective shared libraries when it's opened by `dlopen` on macOS. There's a complicated story having to do with two-level and flat namespaces, common symbols, and changes between Big Sur and Monterey. Using a static library for MPICH circumvents all these problems. The MPIwrapper library is still a shared library, and MPICH is built with `-fPIC` to ensure it can be used in a shared library.